.github/workflows/build-test.yml

```diff
@@ -3,57 +3,101 @@ name: Build and Test
 on: [pull_request]
 
 jobs:
-  build-test:
+  build-test-matrix:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: pkg # applied only for "run" steps
+
     env:
       # We explicitly have this env var not be "CL_DATABASE_URL" to avoid having it be used by core related tests
       # when they should not be using it, while still allowing us to DRY up the setup
       DB_URL: postgresql://postgres:postgres@localhost:5432/chainlink_test?sslmode=disable
+
+    strategy:
+      fail-fast: false
+      matrix:
+        type:
+          # TODO only run when needed
```

Comment:
Should only run based on changes, out of scope for this PR as this was already not the being done